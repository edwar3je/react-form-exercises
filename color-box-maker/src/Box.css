.color-box {
    border: 1px;
    border-color: black;
    margin-left: 10px;
}

.remove-button {
    width: 20px;
    height: 20px;
    color: black;
    background-color: rgb(204, 88, 88);
    border-radius: 10%;
    font-size: 10px;
    margin-left: 10px;
}