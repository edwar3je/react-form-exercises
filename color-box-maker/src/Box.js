import { Fragment } from 'react';
import './Box.css';

const Box = ({height, width, backgroundColor, remove, id}) => {
    
    const styles = {height: height, width: width, backgroundColor: backgroundColor};

    const removeSquare = () => {
        alert(`Removed ${backgroundColor} square that's ${height} high and ${width} wide`);
        remove(id);
    }

    const string = `${backgroundColor}-${height}-${width}`;

    return (
        <Fragment>
            <div className="color-box" id={string} style={styles}></div>
            <button className="remove-button" id={string} onClick={removeSquare}>X</button>
        </Fragment>
    )
}

export default Box;