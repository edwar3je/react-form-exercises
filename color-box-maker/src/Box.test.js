import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Box from './Box';

it("renders without crashing", () => {
    render(
        <Box 
            height= "50px"
            width= "50px"
            backgroundColor= "pink"
            remove= {() => console.log("you clicked the remove button")}
            id="somestring"
        />
    )
});

it("matches the snapshot", () => {
    const { asFragment } = render(
        <Box 
            height= "50px"
            width= "50px"
            backgroundColor= "pink"
            remove= {() => console.log("you clicked the remove button")}
            id="somestring"
        />
    );
    expect(asFragment()).toMatchSnapshot();
});