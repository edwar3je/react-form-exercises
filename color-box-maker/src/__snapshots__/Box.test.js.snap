// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="color-box"
    id="pink-50px-50px"
    style="height: 50px; width: 50px; background-color: pink;"
  />
  <button
    class="remove-button"
    id="pink-50px-50px"
  >
    X
  </button>
</DocumentFragment>
`;
